Packs/VaronisDataSecurityPlatform/Integrations/VaronisDataSecurityPlatform/VaronisDataSecurityPlatform.py

```diff
@@ -0,0 +1,1150 @@
+    :rtype: ``Dict[str, Any]``
+    """
+    builder = SearchEventQueryBuilder(client)
+    builder.create_alert_id_filter(alerts)
+    builder.create_ordering('Event.TimeUTC', 'Asc')
+    query = builder.build()
+    data_range = get_query_range(count, page)
+
+    return execute_search_query(client, query, data_range)
+
+
+''' COMMAND FUNCTIONS '''
+
+
+def test_module(client: Client) -> str:
+    """Tests API connectivity and authentication'
+
+    Returning 'ok' indicates that the integration works like it is supposed to.
+    Connection to the service is successful.
+    Raises exceptions if something goes wrong.
+
+    :type client: ``Client``
+    :param client: client to use
+
+    :return: 'ok' if test passed, anything else will fail the test.
+    :rtype: ``str``
+    """
+
+    message: str = ''
+    try:
+        client.varonis_get_enum(THREAT_MODEL_ENUM_ID)
+        message = 'ok'
+    except DemistoException as e:
+        if 'Unauthorized' in str(e):
+            message = 'Authorization Error: token is incorrect or expired.'
+        else:
+            raise e
+    return message
+
+
+def fetch_incidents(client: Client, last_run: Dict[str, int], first_fetch_time: Optional[datetime], max_results: int,
+                    alert_status: Optional[str], threat_model: Optional[str], severity: Optional[str]
+                    ) -> Tuple[Dict[str, Optional[int]], List[dict]]:
+    """This function retrieves new alerts every interval (default is 1 minute).
+
+    :type client: ``Client``
+    :param client: client to use
+
+    :type last_run: ``Dict[str, int]``
+    :param last_run:
+        A dict with a key containing the latest alert id we got from last fetch
+
+    :type first_fetch_time: ``Optional[datetime]``
+    :param first_fetch_time:
+        If last_run is None (first time we are fetching), it contains
+        the datetime on when to start fetching incidents
+
+    :type max_results: ``int``
+    :param max_results: Maximum numbers of incidents per fetch
+
+    :type alert_status: ``Optional[str]``
+    :param alert_status: status of the alert to search for. Options are 'Open', 'Closed' or 'Under investigation'
+
+    :type threat_model: ``Optional[str]``
+    :param threat_model: Comma-separated list of threat model names of alerts to fetch
+
+    :type severity: ``Optional[str]``
+    :param severity: severity of the alert to search for. Options are 'High', 'Medium' or 'Low'
+
+    :return:
+        A tuple containing two elements:
+            next_run (``Dict[str, Optional[int]]``): Contains last fetched id.
+            incidents (``List[dict]``): List of incidents that will be created in XSOAR
+    :rtype: ``Tuple[Dict[str, int], List[dict]]``
+
+    """
+
+    last_fetched_id = last_run.get('last_fetched_id', None)
+    incidents: List[Dict[str, Any]] = []
+    next_run = {'last_fetched_id': last_fetched_id}
+    iteration: int = 0
+
+    while len(incidents) <= MAX_INCIDENTS_TO_FETCH_PER_RUN:
```

Comment:
It is used for automatic pagination. We are trying to retrieve from api at most 200 incidents and in case it is first run, there could be more for example 1k incidents. In such case we would call api 5 times